private/model/api/passes.go

```diff
@@ -78,6 +78,12 @@ func (r *referenceResolver) resolveReference(ref *ShapeRef) {
 	}
 
 	if shape, ok := r.API.Shapes[ref.ShapeName]; ok {
+		if ref.JSONValue {
+			shape.Type = "jsonvalue"
+			shape.ValueRef.JSONValue = true
+			r.API.Shapes[ref.ShapeName] = shape
```

Comment:
Probably want to create a new Shape here or reuse a shared `JSONValue` shape, because just renaming the existing shape the ref points to could easily cause other refs that point to the same shape to be rendered incorrectly.

In the following model, the shape pointed to by the refs, `A`, and `B` will be converted to a `jsonvalue` type if I'm understanding this code correctly.

```json
"MyShape": {
    "members": {
         "A":  {"shape": "String"},
         "B": {"shape": "String", "JSONValue": true}
    }
}
```